From f745416deac641638c5e363ced2ab17d93002f70 Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Mon, 14 Oct 2013 16:04:58 +0200
Subject: [PORT FROM MAIN] wifi: adapt "wpa_supplicant" to BCM driver "sched_scan"

BZ: 124766

"schedule_scan" is a name that designates different features at
"BCM driver/cfg80211" and "wpa_supplicant" levels.
This patch enables full support of cfg80211 schedule scan through
supplicant's PNO feature.
We don't want to enable supplicant schedule scan that is a different feature.

Change-Id: Ic3ccb259fa24846fd3f63a20c25a0d464ba4989f
Orig-Change-Id: I12b04659566a6b1ba64907c5f25b67d7db05787c
Category: device enablement
Domain: CWS.WIFI-Bluewoods
Origin: upstream
Upstream-Candidate: no,proprietary
Signed-off-by: Jeremie GARCIA <jeremiex.garcia@intel.com>
---
 src/drivers/driver_nl80211.c |   13 ++-----------
 1 files changed, 2 insertions(+), 11 deletions(-)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index 3434647..e8f7dc9 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -3069,7 +3069,8 @@ static void wiphy_info_supp_cmds(struct wiphy_info_data *info,
 			info->connect_supported = 1;
 			break;
 		case NL80211_CMD_START_SCHED_SCAN:
-			info->capa->sched_scan_supported = 1;
+                        /* BCM workaround (patch 119707) */
+			info->capa->sched_scan_supported = 0;
 			break;
 		case NL80211_CMD_PROBE_CLIENT:
 			info->poll_command_supported = 1;
@@ -4423,11 +4424,6 @@ static int wpa_driver_nl80211_sched_scan(void *priv,
 
 	wpa_dbg(drv->ctx, MSG_DEBUG, "nl80211: sched_scan request");
 
-#ifdef ANDROID
-	if (!drv->capa.sched_scan_supported)
-		return android_pno_start(bss, params);
-#endif /* ANDROID */
-
 	msg = nl80211_scan_common(drv, NL80211_CMD_START_SCHED_SCAN, params,
 				  bss->wdev_id_set ? &bss->wdev_id : NULL);
 	if (!msg)
@@ -4508,11 +4504,6 @@ static int wpa_driver_nl80211_stop_sched_scan(void *priv)
 	int ret = 0;
 	struct nl_msg *msg;
 
-#ifdef ANDROID
-	if (!drv->capa.sched_scan_supported)
-		return android_pno_stop(bss);
-#endif /* ANDROID */
-
 	msg = nlmsg_alloc();
 	if (!msg)
 		return -1;
-- 
1.7.4.1

